api/app/serializers/v1/resource_import_serializer.rb

```diff
@@ -4,22 +4,6 @@ class ResourceImportSerializer < ManifoldSerializer
     include ::V1::Concerns::ManifoldSerializer
 
     typed_attribute :source, Types::String.enum("google_sheet", "attached_data")
-    typed_attribute :data, Types::Hash.schema(
```

Comment:
This is a weird case in that it's handling uploads differently from other parts of the application. Take an "avatar" upload field, for example. Normally the request would include something like this in the avatar attribute:
```
    {
      content_type: text/csv,
      data: "data:text/csv;base64,foobar",
      filename: resources.csv
    }
```
And the response would include `avatar_styles` with info about the avatars.

In this case, we're using attributes.data for the request _and_ the response, which makes things hard because the factories are mostly setup to generate a structure that we can look at for the response, not the request. 

There are a few things I've done to address this:

1. Refactored the 'param_helpers' methods. They were messy and hard to understand with poor method names.
2. Made the methods context sensitive by including a `type` keyword argument that can be :request or :response
3. If we're generating a json request structure, the helper will loop through the attributes and see if any of the values are file-like. If they are, it will replace the object with a suitable request datastructure.
4. I've also removed the `data` attribute from the resource import serializer response. It is never used in the client application, and trying to support the same property that looks one way in the request and another way in the response is more work than I want to tackle right now.